packages/esm-appointments-app/src/form/appointments-form.resource.ts

```diff
@@ -0,0 +1,126 @@
+  */
+  const fetcher = () =>
+    openmrsFetch(appointmentsSearchUrl, {
+      method: 'POST',
+      signal: abortController.signal,
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: {
+        patientUuid: patientUuid,
+        startDate: startDate,
+      },
+    });
+
+  const { data, error, isLoading, isValidating, mutate } = useSWR<AppointmentsFetchResponse, Error>(
+    appointmentsSearchUrl,
+    fetcher,
+  );
+
+  const appointments = data?.data?.length ? data.data : null;
+
+  const pastAppointments = appointments
+    ?.sort((a, b) => (b.startDateTime > a.startDateTime ? 1 : -1))
+    ?.filter(({ status }) => status !== 'Cancelled')
+    ?.filter(({ startDateTime }) =>
+      dayjs(new Date(startDateTime).toISOString()).isBefore(new Date().setHours(0, 0, 0, 0)),
+    );
+
+  const upcomingAppointments = appointments
+    ?.sort((a, b) => (a.startDateTime > b.startDateTime ? 1 : -1))
+    ?.filter(({ status }) => status !== 'Cancelled')
+    ?.filter(({ startDateTime }) => dayjs(new Date(startDateTime).toISOString()).isAfter(new Date()));
+
+  const todaysAppointments = appointments
+    ?.sort((a, b) => (a.startDateTime > b.startDateTime ? 1 : -1))
+    ?.filter(({ status }) => status !== 'Cancelled')
+    ?.filter(({ startDateTime }) => dayjs(new Date(startDateTime).toISOString()).isToday());
+
+  return {
+    data: data ? { pastAppointments, upcomingAppointments, todaysAppointments } : null,
+    isError: error,
+    isLoading,
+    isValidating,
+    mutate,
+  };
+}
+
+export function useAppointmentService() {
+  const { data, error, isLoading } = useSWR<{ data: Array<AppointmentService> }, Error>(
+    `/ws/rest/v1/appointmentService/all/full`,
+    openmrsFetch,
+  );
+
+  return {
+    data: data ? data.data : null,
+    isError: error,
+    isLoading,
+  };
+}
+
+export function saveAppointment(appointment: AppointmentPayload, abortController: AbortController) {
+  return openmrsFetch(`/ws/rest/v1/appointment`, {
+    method: 'POST',
+    signal: abortController.signal,
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: appointment,
+  });
+}
+
+export function saveRecurringAppointments(
+  recurringAppointments: RecurringAppointmentsPayload,
+  abortController: AbortController,
+) {
+  return openmrsFetch(`/ws/rest/v1/recurring-appointments`, {
+    method: 'POST',
+    signal: abortController.signal,
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: recurringAppointments,
+  });
+}
+
+export function getAppointmentsByUuid(appointmentUuid: string, abortController: AbortController) {
+  return openmrsFetch(`/ws/rest/v1/appointments/${appointmentUuid}`, {
+    signal: abortController.signal,
+  });
+}
+
+export function getAppointmentService(abortController: AbortController, uuid) {
+  return openmrsFetch(`/ws/rest/v1/appointmentService?uuid=` + uuid, {
+    signal: abortController.signal,
+  });
+}
```

Comment:
I haven't tracked the context in which these are used, but can they be re-written to take advantage of SWR?

Reply:
Code I copied over, I'm not sure where they are used as well... I hope to do a more thorough cleanup of all the resource calls in the future, I'll had a TODO to the code to take a look at that then.